### ISSUE TITLE
Is MSAL ready for hybrid and postquantum PKI?

### ISSUE THREAD

**Issue by 3BK** [AzureAD/microsoft-authentication-library-for-dotnet#3848]
It appears that MSAL uses algorithms like RS256.

Is MSAL ready for hybrid and postquantum PKI?

references
https://www.ietf.org/archive/id/draft-prorock-cose-post-quantum-signatures-01.txt
https://dl.acm.org/doi/abs/10.1007/978-3-031-20974-1_20
https://paragonie.com/blog/2017/03/jwt-json-web-tokens-is-bad-standard-that-everyone-should-avoid

**Comment by SameerK-MSFT** [comment]
Interesting question. @SaeedAkhter-MSFT do you have any input on this? Thx

**Comment by bgavrilMS** [comment]
Afaik AAD only supports RSA and is building support for ECD certs, so no, no post-quantum yet.

**Comment by 3BK** [comment]
> AAD is building support for ECD certs,
That's hopefully a step in the right direction - towards hybrid PKI

**Comment by 3BK** [comment]
Don't forget to refresh your hardware roadmap. ;)

**Comment by bgavrilMS** [comment]
Well, it's not just AAD. Clients (web sites, web apis) need to have new crypto stacks to be able to request tokens using signed assertions (a signed assertion is just another JWT, but one that the client generates). And RPs need to have it as well, to be able to verify signatures....

**Comment by 3BK** [comment]
Agreed.  Hybrid (and/or post quantum) are substantial changes.  (Probably driven by MS Balance Scorecard.)

**Comment by bgavrilMS** [comment]
PostQuantum updates will be issued by Entra directly, this is an area we are looking at holistically. 

Strictly from MSAL perspective, only the client assertion creation use a crypto algorithm (RSA), as well as a the creation of SHR POP tokens.
